### PR TITLE
[engine] Sync Flutter 3.10 source code

### PR DESCRIPTION
**Pull request by swift-kim**
Copy `embedder.h` and `client_wrapper` from the Flutter 3.10 engine source tree.

### CODE REVIEW

**Comment by swift-kim** [comment]
I'm going to create two more PRs that depend on this PR. I'll merge this PR without approval if there's no review by next Monday. 

Please also review my recent PRs in this and other repos so that we can finish the migration to Flutter 3.10 quickly.

File: flutter/shell/platform/common/client_wrapper/include/flutter/event_stream_handler.h
Comment (bbrto21) [inline flutter/shell/platform/common/client_wrapper/include/flutter/event_stream_handler.h:17]: This problem is finally solved!

**Review by bbrto21 — APPROVED** [review]
LGTM

File: flutter/shell/platform/common/client_wrapper/include/flutter/event_stream_handler.h
Comment (swift-kim) [reply flutter/shell/platform/common/client_wrapper/include/flutter/event_stream_handler.h:17]: Yes, the workaround you applied to the plugins can now be removed but this change will not break the plugins since there's no plugin that utilizes the `error_details` field.

**Pull request merged**
